### PR TITLE
[terra-clinical-item-display] Default text style set to primary.

### LINKED ISSUE #526: [clinical-item-display] Make the text style primary by default
# Feature Request

## Description

We should make the text style primary by default in the next MVB

### LINKED ISSUE #673: [terra-clinical-item-view] insufficient color-contrast for secondary content on hover/focus background
# Bug Report

## Description
When a secondary item view content  is placed in a terra-list there is insufficient color-contrast (only for fusion-theme) between color of a secondary item view content and the selected hover/focus background of the list. This is happening in Firefox/IE but works in Chrome.

The fusion-theme color for [--terra-clinical-item-view-content-secondary-color](https://github.com/cerner/terra-clinical/blob/821a6f560c7ce0f9ab2fb50f101d198506822313/packages/terra-clinical-item-view/src/orion-fusion-theme/ItemView.module.scss#L5) is `#64696c`. The fusion-theme for [--terra-list-item-selected-focus-background-color](https://github.com/cerner/terra-core/blob/e7ca384ee08d77ec69574ca2682f403676468cf1/packages/terra-list/src/orion-fusion-theme/List.module.scss#L28-L29) is `#cbe7fa`. The contrast ratio for these colors is `4.32 : 1`. 

## Steps to Reproduce

1. Go to the [selectable item collection test](https://engineering.cerner.com/terra-clinical/tests/terra-clinical-item-collection/clinical-item-collection/item-collection-selectable).
2. Resize to a small breakpoint to show the list view.
3. Select and hover over the first item.
4. Inspect the secondary content ("Display2") and its background.

## Additional Context / Screenshots
Screen Shot 2020-06-12 at 3 58 18 PM

```
1) [medium] Item Collection Returns child key for click when onSelect is given to a selectable item collection [default] is accessible and is within the mismatch tolerance:
expected no accessibility violations but got:
	{
  "description": "Ensures the contrast between foreground and background colors meets WCAG 2 AA contrast ratio thresholds",
  "help": "Elements must have sufficient color contrast",
  "helpUrl": "https://dequeuniversity.com/rules/axe/3.5/color-contrast?application=axeAPI",
  "id": "color-contrast",
  "impact": "serious",
  "nodes": [
    {
      "all": [],
      "any": [
        {
          "data": {
            "bgColor": "#cbe7fa",
            "contrastRatio": 4.32,
            "expectedContrastRatio": "4.5:1",
            "fgColor": "#64696c",
            "fontSize": "10.5pt (14px)",
            "fontWeight": "normal",
            "messageKey": null
          },
          "id": "color-contrast",
          "impact": "serious",
          "message": "Element has insufficient color contrast of 4.32 (foreground color: #64696c, background color: #cbe7fa, font size: 10.5pt (14px), font weight: normal). Expected contrast ratio of 4.5:1",
          "relatedNodes": [
            {
              "html": "<li tabindex=\"0\" id=\"item1\" role=\"option\" aria-selected=\"true\" data-item-show-focus=\"true\" class=\"List-module__list-item___1Gt4b List-module__selected___32shv List-module__is-selectable___3PyDe\">",
              "target": [
                "#item1"
              ]
            }
          ]
        }
      ],
      "failureSummary": "Fix any of the following:\n  Element has insufficient color contrast of 4.32 (foreground color: #64696c, background color: #cbe7fa, font size: 10.5pt (14px), font weight: normal). Expected contrast ratio of 4.5:1",
      "html": "<div data-terra-clinical-item-display-text=\"true\" class=\"ItemDisplay-module__text___c-QUG\">Display 2</div>",
      "impact": "serious",
      "none": [],
      "target": [
        "#item1 > .List-module__item-fill___3VrIn > .ItemView-module__item-view___61CR3.ItemView-module__one-column___2LPcY > .ItemView-module__body___SD0Vl > .ItemView-module__row-container___3YCG9 > .ItemView-module__row___1_JaL:nth-child(3) > .ItemView-module__content-secondary-color___3j2Dv.ItemView-module__content-secondary-size___2MzM9.ItemView-module__content___3wPtn > .ItemDisplay-module__item-display___3yi9q > .ItemDisplay-module__text___c-QUG[data-terra-clinical-item-display-text=\"true\"]"
      ]
    }
  ],
  "tags": [
    "cat.color",
    "wcag2aa",
    "wcag143"
  ]
}
```

## Environment

* Component Name and Version:
* Browser Name and Version:
* Node/npm Version: [e.g. Node 8/npm 5]
* Webpack Version:
* Operating System and version (desktop or mobile):

### PR DESCRIPTION
**Pull request by saket2403**
### Summary
Resolves #526 
Resolves #673

### CODE REVIEW

File: packages/terra-clinical-item-display/src/ItemDisplay.jsx [hunk 1]
```diff
@@ -47,8 +47,7 @@ const propTypes = {
 
 const defaultProps = {
   text: '',
-  // TODO: textStyle should be set to 'primary' on the next major version bump: https://github.com/cerner/terra-clinical/issues/526
-  textStyle: undefined,
+  textStyle: 'primary',
```
Comment (emilyrohrbough) [inline packages/terra-clinical-item-display/src/ItemDisplay.jsx]: We are good with making this a MVB now?

Comment (saket2403) [reply packages/terra-clinical-item-display/src/ItemDisplay.jsx]: @bjankord .Please have a look.

File: packages/terra-clinical-item-collection/CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,7 @@ Unreleased
 4.6.0 - (August 14, 2019)
 ----------
 ### Changed
+* Jest snapshot updated for ItemDisplay's primary textStyle default prop.
```
Comment (StephenEsser) [inline packages/terra-clinical-item-collection/CHANGELOG.md]: Reminder to move this into an unreleased section

File: packages/terra-clinical-item-display/src/ItemDisplay.jsx (hunk 1 above)
Comment (StephenEsser) [inline packages/terra-clinical-item-display/src/ItemDisplay.jsx]: How would a user revert back to the previous default / un-styled text? Or is our intention with this change to discourage any use of un-styled text?

For example, a user with a component prior to this change that looks like this would display with the default un-styled text.

```jsx
<ItemDisplay text="Example Display" />
```

This example after this change would update to primary text. To revert back I believe a user would need to do something like this which feels a little strange:

```jsx
<ItemDisplay text="Example Display" textStyle='' />
```


cc: @neilpfeiffer Do we want to discourage any use of the ItemDisplay with un-styled text, do we want to provide a new textStyle?

Comment (jeremyfuksa) [reply packages/terra-clinical-item-display/src/ItemDisplay.jsx]: It would be my opinion that we would not want a consumer using unstyled text anywhere.

Comment (neilpfeiffer) [reply packages/terra-clinical-item-display/src/ItemDisplay.jsx]: It technically doesn't matter (if you only examine this from the prop standpoint, it seems like a big deal but in actuality it isn't) -- since the style of `undefined` is/was/has-been exactly the same as `primary`--> this is mainly a fix to conceptualization of the options and clearer/cleaner with documenting in the propsTable (after change will show the default is 'primary') which then matches the examples above, and makes more sense when we spell out the options 'Primary', 'Secondary', 'Attention', etc. 

It was odd when there was a `'secondary'` without a `'primary'` (undefined has always equaled primary, and was called out in css classes and examples, but not as prop), so it was added, and now it is odd having undefined and `'primary'` both doing the same thing. This MVB will not visually/wdio affect anyone who has currently not set a `textStyle`, and won't affect anyone who has - it simply is for better documentation and understanding of the text options available (which might also eventually get put into other components).

To @jeremyfuksa's point - there really is no reason to unset this, because unsetting is the same as setting it to primary text (our default styling). Nor would 'undefined' ever look differently and need a new/another style.

And to echo @StephenEsser -- `textStyle=''` is just strange (...though I was thinking of another word).

**Comment by neilpfeiffer** [comment]
Need to update the examples:
for **Item Display: Text styles** -- remove 'Default (same as Primary)'
for **Disabled Item Display: Text styles** -- remove 'Default'

**Comment by saket2403** [comment]
> Need to update the examples:
> for **Item Display: Text styles** -- remove 'Default (same as Primary)'
> for **Disabled Item Display: Text styles** -- remove 'Default'

Updated 49f0f52

File: packages/terra-clinical-item-display/src/terra-dev-site/doc/example/TextStyles.jsx [hunk 3]
```diff
@@ -3,7 +3,6 @@ import ItemDisplay from 'terra-clinical-item-display';
 
 const component = () => (
   <React.Fragment>
-    <ItemDisplay text="Default (same as Primary)" />
     <ItemDisplay text="Primary" textStyle="primary" />
```
Comment (neilpfeiffer) [inline packages/terra-clinical-item-display/src/terra-dev-site/doc/example/TextStyles.jsx:6]: Lets also update this, for those looking at code examples to not think they need to add a textStyle:
```suggestion
    <ItemDisplay text="Primary" />
```

File: packages/terra-clinical-item-display/src/terra-dev-site/doc/example/TextStylesDisabled.jsx [hunk 4]
```diff
@@ -3,7 +3,6 @@ import ItemDisplay from 'terra-clinical-item-display';
 
 const component = () => (
   <React.Fragment>
-    <ItemDisplay text="Default" isDisabled />
     <ItemDisplay text="Primary" textStyle="primary" isDisabled />
```
Comment (neilpfeiffer) [inline packages/terra-clinical-item-display/src/terra-dev-site/doc/example/TextStylesDisabled.jsx:6]: Lets also update this, for those looking at code examples to not think they need to add a textStyle:

```suggestion
    <ItemDisplay text="Primary" isDisabled />
```

File: packages/terra-clinical-item-display/src/terra-dev-site/doc/example/TextStyles.jsx (hunk 3 above)
Comment (saket2403) [reply packages/terra-clinical-item-display/src/terra-dev-site/doc/example/TextStyles.jsx]: Updated

File: packages/terra-clinical-item-display/src/terra-dev-site/doc/example/TextStylesDisabled.jsx (hunk 4 above)
Comment (saket2403) [reply packages/terra-clinical-item-display/src/terra-dev-site/doc/example/TextStylesDisabled.jsx]: Updated

**Comment by neilpfeiffer** [comment]
UX Review in progress

**Comment by neilpfeiffer** [comment]
@saket2403 -- after reviewing the updates to snapshots, particularly noticeable in the themes, by adding the primary variant as default, it adds the primary class, which the intent is to simply use the base text color, but because the variable is unvalued in default theme but valued in the other themes, we missed that in the themes - this extra class adds higher specificity that overrides the text color set for a comment, or the secondary displays in the clinical-item-view. 

---
We will need to make some additional updates:

1. Update how the text style classes are being applied, so that if Primary, no class is added.
https://github.com/cerner/terra-clinical/blob/9e6b74155ba3f6b1e4cc310e51aa609e975b34b2/packages/terra-clinical-item-display/src/ItemDisplay.jsx#L72-L76
Something like:
```
  const textClassNames = cx([
    'text',
    { 'is-truncated': isTruncated },
    { [`${textStyle}`]: textStyle === TextStyles.SECONDARY },
    { [`${textStyle}`]: textStyle === TextStyles.ATTENTION },
    { [`${textStyle}`]: textStyle === TextStyles.STRONG },
    { 'strike-through': textStyle === TextStyles.STRIKETHROUGH },
  ]);

```

---
2. Remove the `.primary` class and the `...-primary-color` variable in all the themes:
https://github.com/cerner/terra-clinical/blob/9e6b74155ba3f6b1e4cc310e51aa609e975b34b2/packages/terra-clinical-item-display/src/ItemDisplay.module.scss#L60-L62
https://github.com/cerner/terra-clinical/blob/9e6b74155ba3f6b1e4cc310e51aa609e975b34b2/packages/terra-clinical-item-display/themes/clinical-lowlight-theme/clinical-lowlight-theme.scss#L5
https://github.com/cerner/terra-clinical/blob/9e6b74155ba3f6b1e4cc310e51aa609e975b34b2/packages/terra-clinical-item-display/themes/orion-fusion-theme/orion-fusion-theme.scss#L7

---
3. Update the Strong class to have a color:
https://github.com/cerner/terra-clinical/blob/9e6b74155ba3f6b1e4cc310e51aa609e975b34b2/packages/terra-clinical-item-display/src/ItemDisplay.module.scss#L73-L76
```
  .strong {
    color: var(--terra-clinical-item-display-strong-color, #1c1f121);
    font-weight: var(--terra-clinical-item-display-strong-font-weight, 700);
  }
```
and fix lowlight-theme to have a color:
https://github.com/cerner/terra-clinical/blob/9e6b74155ba3f6b1e4cc310e51aa609e975b34b2/packages/terra-clinical-item-display/themes/clinical-lowlight-theme/clinical-lowlight-theme.scss#L7
```
  --terra-clinical-item-display-strong-color: #b2b5b6;
```

---
4. Update the StrikeThrough class to have a color:
https://github.com/cerner/terra-clinical/blob/9e6b74155ba3f6b1e4cc310e51aa609e975b34b2/packages/terra-clinical-item-display/src/ItemDisplay.module.scss#L78-L81
```
  .strike-through {
    color: var(--terra-clinical-item-display-strike-through-color, #1c1f121);
    text-decoration: line-through;
  }
```
and fix lowlight-theme to have a color:
https://github.com/cerner/terra-clinical/blob/9e6b74155ba3f6b1e4cc310e51aa609e975b34b2/packages/terra-clinical-item-display/themes/clinical-lowlight-theme/clinical-lowlight-theme.scss#L8
```
  --terra-clinical-item-display-strike-through-color: #b2b5b6;
```

---
5. Add these missing variables & values to the lowlight-theme:
```
  --terra-clinical-item-display-attention-font-weight: 700;
  --terra-clinical-item-display-strong-font-weight: 700;
```

---
6. Update the Secondary colors to be RGBA (somehow got missed when the item-display was updated previously):
https://github.com/cerner/terra-clinical/blob/9e6b74155ba3f6b1e4cc310e51aa609e975b34b2/packages/terra-clinical-item-view/src/ItemView.module.scss#L61
```
    color: var(--terra-clinical-item-view-content-color, rgba(28, 31, 33, 0.65));
```

https://github.com/cerner/terra-clinical/blob/9e6b74155ba3f6b1e4cc310e51aa609e975b34b2/packages/terra-clinical-item-view/src/ItemView.module.scss#L112
```
    color: var(--terra-clinical-item-view-content-secondary-color, rgba(28, 31, 33, 0.65));
```
and in fusion-theme:
https://github.com/cerner/terra-clinical/blob/9e6b74155ba3f6b1e4cc310e51aa609e975b34b2/packages/terra-clinical-item-view/themes/orion-fusion-theme/orion-fusion-theme.scss#L4-L6
```
  --terra-clinical-item-view-content-color: rgba(28, 31, 33, 0.65);
  --terra-clinical-item-view-content-primary-color: #1c1f21;
  --terra-clinical-item-view-content-secondary-color: rgba(28, 31, 33, 0.65);
```
---

After making these changes, we should expect to no longer see changes in the lowlight-theme and fusion-theme screenshots for item-view and the comment version of item-display.

**Comment by neilpfeiffer** [comment]
cc: @PaulSolDev - the changes to colors in the comment above will be something you will want to update in your documented requirements (changes from solid HEX color to RGBA with opacity, but it should not change the final visual appearances).

**Comment by PaulSolDev** [comment]
Thank you, @neilpfeiffer . I'll  get something logged on my end and hopefully sent out today to reflect these updates.

**Comment by saket2403** [comment]
> Changes: [#534 (comment)](https://github.com/cerner/terra-clinical/pull/534#issuecomment-638872302)

Done here cb53a4d .

File: packages/terra-clinical-item-view/src/ItemView.module.scss [hunk 5]
```diff
@@ -113,7 +113,7 @@
   }
 
   .content-secondary-color {
-    color: var(--terra-clinical-item-view-content-secondary-color, #64696c);
+    color: var(--terra-clinical-item-view-content-secondary-color, rgba(28, 31, 33, 0.65));
```
Comment (benbcai) [inline packages/terra-clinical-item-view/src/ItemView.module.scss:116]: Since this will resolve #673 could you revert the change for disabling the color-contrast in this test? https://github.com/cerner/terra-clinical/pull/674/files#diff-9ee238e41af23d6170088980ae1de2f5R34-R35 ?

File: packages/terra-clinical-item-collection/CHANGELOG.md [hunk 6]
```diff
@@ -32,6 +32,7 @@ Unreleased
 4.19.0 - (April 14, 2020)
 ----------
 ### Changed
+* Jest and wdio snapshot updated for ItemDisplay's primary textStyle default prop.
```
Comment (emilyrohrbough) [inline packages/terra-clinical-item-collection/CHANGELOG.md:35]: this need to be updated to the unreleased section

File: packages/terra-clinical-item-view/src/ItemView.module.scss (hunk 5 above)
Comment (saket2403) [reply packages/terra-clinical-item-view/src/ItemView.module.scss:116]: Reverted 87f5525

File: packages/terra-clinical-item-collection/CHANGELOG.md (hunk 6 above)
Comment (saket2403) [reply packages/terra-clinical-item-collection/CHANGELOG.md]: Moved 87f5525

**Comment by ryanthemanuel** [comment]
Spoke with @PaulSolDev and he was ok with this being reviewed just by UX.
